Script/Commands: Cast minor changes and checks

duhow opened the pull request:
Some functions had no checks for **IsSpellValid**, or just **returned false**, which doesn't show any message on client.

Minor change on Cast Self. As it is **self**, you don't need to have a **target**.

duhow committed:
Command Cast minor changes and checks

Some functions had no checks for IsSpellValid, or just returned false, which doesn't show any message on client.
Minor change on Cast Self. As it is self, you don't need to have a target.

joschiwald commented on src/server/scripts/Commands/cs_cast.cpp:
```diff
@@ -191,13 +202,7 @@ class cast_commandscript : public CommandScript
-            handler->SetSentErrorMessage(true);
-            return false;
-        }
+        Unit* player = handler->GetSession()->GetPlayer();
```
that is not a good idea, you should be able to select another creature and let him cast the spell

duhow replied:
That makes no sense if it is **self cast**, otherwise it would be **cast (to victim)** or **cast back**...

joschiwald replied:
it makes sense if you want the target casts a spell to itself
.cast : player -> target
.cast back : target -> player
.cast target : target -> target's victim
.cast self : target -> target

joschiwald commented:
you could create a helper function to avoid duplicate code

duhow committed:
Rebuild again

Rebuild again

tkrokli commented:
I wonder if Travis fails because `duhow:patch-2` is not updated. I see `AnimKitID` mentioned in both fatal errors, whereas `AnimKitID` was added to src/server/game/Server/Packets/MiscPackets.h ≈ 4 hours ago.

duhow commented:
Yup, I think the same. That's why I reposted again an upload...
El 13/7/2015 2:39, "Trond B Krokli" <notifications@github.com> escribió:

> I wonder if Travis fails because duhow:patch-2 is not updated. I see
> AnimKitID mentioned in both fatal errors, whereas AnimKitID was added to
> src/server/game/Server/Packets/MiscPackets.h ≈ 4 hours ago.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TrinityCore/TrinityCore/pull/15061#issuecomment-120776531>
> .
>

jackpoz commented:
manually restarted travis build and it passed

duhow committed:
Change condition for self cast

First, use selected unit as target. If no target selected, use player (self).
Finally, cast spell to self target.

jackpoz commented on src/server/scripts/Commands/cs_cast.cpp:
```diff
@@ -194,9 +204,8 @@ class cast_commandscript : public CommandScript
         Unit* target = handler->getSelectedUnit();
```
please add a new method "getSelectedUnitOrSelf()" just like there is already "getSelectedPlayerOrSelf()" so return the selected unit or session player

jackpoz commented on src/server/scripts/Commands/cs_cast.cpp:
```diff
@@ -194,9 +204,8 @@ class cast_commandscript : public CommandScript
-            handler->SetSentErrorMessage(true);
-            return false;
+            // If no target selected, then target is player.
+            target = handler->GetSession()->GetPlayer();
```
Actually this change is totally useless because https://github.com/TrinityCore/TrinityCore/blob/2e586bc520e7eda2387e850c3c0049e6daa880de/src/server/game/Chat/Chat.cpp#L617 already returns the session player if there is no Unit*, did you check how getSelectedUnit() works before modifying the code and debugged what happens when no target is selected ingame ?
